### PR TITLE
Add a section about ElastiCache, fixes #349

### LINKED ISSUE #349: New section: ElastiCache

### PR DESCRIPTION
**Pull request by cristim**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1154,6 +1155,41 @@ RDS SQL Server
...
+- Choose the instance type carefully based on your caching needs, since it can't
+  be changed later on an existing cluster(horizontal scaling by changing the
+  number of nodes is supported). You will need to launch a new cluster with the
+  new instance type, and reconfigure your application to point to it, then
+  delete the old cluster.
+
+### ElastiCache Gotchas and Limitations
+
+- Since clusters can't be scaled vertically by changing the instance type, it
+  may become a problem if they were launched using CloudFormation in as stack
```
Comment (abrussak) [inline README.md]: `in as stack` -> `in a stack`

File: README.md [hunk 2]
```diff
@@ -1154,6 +1155,41 @@ RDS SQL Server
...
+- 📒 [Homepage](https://aws.amazon.com/elasticache/) ∙ [User
+  guide](http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide) ∙
+  [FAQ](https://aws.amazon.com/elasticache/faqs/) ∙
+  [Pricing](https://aws.amazon.com/elasticathe/pricing/)
+- **ElastiCache** is a managed in-memory cache service, that can be used to
+  store temporary data in a fast in-memory cache, typically in order to avoid
+  repeating the same computation multiple times when it could be reused. It
+  supports both the [Memcached](https://memcached.org) and
+  [Redis](https://redis.io) open source in-memory cache software.
+- The main benefit is that AWS takes care of running, patching and optimizing
```
Comment (jlevy) [inline README.md]: Ideally we give some guidance about when one should or shouldn't use ElastiCache. Can we say anything about the reasons you might not want to use AWS' managed deployments?

Are there only specific versions for features supported, possible incompatibilities, etc? (For example, by analogy, Postgres RDS only supports certain plugins.)

File: README.md [hunk 3]
```diff
@@ -1154,6 +1155,41 @@ RDS SQL Server
...
+  delete the old cluster.
+
+### ElastiCache Gotchas and Limitations
+
+- Since clusters can't be scaled vertically by changing the instance type, it
+  may become a problem if they were launched using CloudFormation in as stack
+  that also contains other resources. Because of this, it is recommended to
+  Launch each ElastiCache cluster in its dedicated CloudFormation stack.
+
+
```
Comment (jlevy) [inline README.md]: Until this is more complete, let's add a standard disclaimer:

🚧 [*Please help expand this incomplete section.*](CONTRIBUTING.md)

File: README.md [hunk 4]
```diff
@@ -1154,6 +1155,41 @@ RDS SQL Server
...
+  repeating the same computation multiple times when it could be reused. It
+  supports both the [Memcached](https://memcached.org) and
+  [Redis](https://redis.io) open source in-memory cache software.
+- The main benefit is that AWS takes care of running, patching and optimizing
+  the cache nodes for you, so you just need to launch it and configure an
+  endpoint in your application, while AWS will take of all the operational work.
+
+### ElastiCache Tips
+
+- Choose the instance type carefully based on your caching needs, since it can't
```
Comment (jlevy) [inline README.md]: Any links to add here that would be helpful?

File: README.md [hunk 5]
```diff
@@ -1154,6 +1155,41 @@ RDS SQL Server
...
+  number of nodes is supported). You will need to launch a new cluster with the
+  new instance type, and reconfigure your application to point to it, then
+  delete the old cluster.
+
+### ElastiCache Gotchas and Limitations
+
+- Since clusters can't be scaled vertically by changing the instance type, it
+  may become a problem if they were launched using CloudFormation in as stack
+  that also contains other resources. Because of this, it is recommended to
+  Launch each ElastiCache cluster in its dedicated CloudFormation stack.
```
Comment (jlevy) [inline README.md]: Nits: Launch -> launch
its dedicated -> a dedicated

**Comment by cristim** [comment]
Sorry, I messed up a bit, and not all fixes are implemented yet, stay tuned.

I'm having a few issues isolating this from #360 (git interactive add is hard to use continuously), I will be able to submit a better patch once #360 is merged.

File: README.md [hunk 6]
```diff
@@ -1159,6 +1160,48 @@ RDS SQL Server
...
+
+- Choose the
+  [engine](http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/SelectEngine.Uses.html),
+  clustering configuration and [instance
+  type](http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/CacheNodes.SelectSize.html)
+  carefully based on your application needs, since in many cases it can't be
+  changed later on an existing cluster. The documentation explains in detail the
+  pros, cons and limitations of each of them in order to help you choose the
+  best fit for your application.
+- In most cases Redis is a better choice than memcached.
```
Comment (max-zanko) [inline README.md]: This statement is controversial and needs to be backed by facts and links to credible sources.

File: README.md [hunk 7]
```diff
@@ -1159,6 +1160,48 @@ RDS SQL Server
...
+
+### ElastiCache Gotchas and Limitations
+
+- Since in many cases the cache clusters can't be changed easily, it may become
+  a problem if they were launched using CloudFormation in a stack that also
+  contains other resources and you really need to change the cache. Because of
+  this, it is recommended to Launch each ElastiCache cluster (just like any other
```
Comment (max-zanko) [inline README.md]: s/Launch/launch/

Comment (max-zanko) [inline README.md]: "change the cache" -> "change the cache cluster" for clarity?

File: README.md [hunk 8]
```diff
@@ -1159,6 +1160,48 @@ RDS SQL Server
...
+  [FAQ](https://aws.amazon.com/elasticache/faqs/) ∙
+  [Pricing](https://aws.amazon.com/elasticathe/pricing/)
+- **ElastiCache** is a managed in-memory cache service, that can be used to
+  store temporary data in a fast in-memory cache, typically in order to avoid
+  repeating the same computation multiple times when it could be reused.
+- It supports both the [Memcached](https://memcached.org) and
+  [Redis](https://redis.io) open source in-memory cache software.
+- The main benefit is that AWS takes care of running, patching and optimizing
+  the cache nodes for you, so you just need to launch a cluster and configure
+  its endpoint in your application, while AWS will take of all the operational
```
Comment (max-zanko) [inline README.md]: s/take/take care/
Also, I'd mild it down a bit: "AWS will take care of most of the operational work of running the cluster"

File: README.md (hunk 6 above)
Comment (DavidTPate) [inline README.md]: Yeah, I would just remove it. Talking about the different cases that they serve might be handy. Memcached for example can be run in a hash-ring.

**Comment by JannKleen** [comment]
Maybe add a note about ElastiCache not supporting authentication, so it's not wise to use it as a drop-in Redis replacement for Services that don't run in your AWS account

**Comment by QuinnyPig** [comment]
Poke. Can we get this merged in soon? Great content, don't want to see it languish.

**Comment by QuinnyPig** [comment]
@cristim Anything I can do to help with this?

**Comment by cristim** [comment]
@QuinnyPig a review of the current change would be nice, I did my best to address all previous comments.

**Comment by QuinnyPig** [comment]
Man, one of these days I'll learn GitHub's interface. Sorry about that!

**Pull request merged**